firedrake/slate/slac/compiler.py

```diff
@@ -290,13 +583,17 @@ def eigen_matrixbase_type(shape):
              library syntax.
     """
     if len(shape) == 0:
-        raise NotImplementedError("Scalar-valued expressions cannot be declared as an Eigen::MatrixBase object.")
+        raise NotImplementedError(
+            "Scalar exprs cannot be declared as an Eigen::MatrixBase object."
```

Comment:
I have tried this before, and I couldn't quite get it to work (I did precisely as you said). So at the moment I am raising `NotImplementedError` if one tries to assemble a zero-rank tensor.